BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -568,3 +533,25 @@ struct Measurer: Measurable {
         _measure(constraint)
     }
 }
+
+
+extension Array {
+
+    /// A `map` implementation that also passes the `index` of each element in the original array.
+    ///
+    /// This method is more performant than calling `array.enumerated().map(...)` by up
+    /// to 25% for large collections, so prefer it when needing an indexed `map` in areas where performance is critical.
+    @inlinable func indexedMap<Mapped>(_ map: (Int, Element) -> Mapped) -> [Mapped] {
+
+        let count = self.count
+
+        var mapped = [Mapped]()
+        mapped.reserveCapacity(count)
+
+        for index in 0..<count {
```

Comment:
(Testing w/ `test_deep_element_hierarchy` since it exercises this codepath more)

`self.indices` tries:
Iterations: 30, Average Time: 0.3376573999722799
Iterations: 30, Average Time: 0.33470939795176186
Iterations: 30, Average Time: 0.33426090081532794

`0<...count` tries:
Iterations: 30, Average Time: 0.33422886530558266
Iterations: 30, Average Time: 0.3350138028462728
Iterations: 30, Average Time: 0.33537230094273884

So, doesn't matter! Switching to this.